src/payment/strategies/klarna/klarna-payment-strategy.spec.ts

```diff
@@ -118,12 +126,49 @@ describe('KlarnaPaymentStrategy', () => {
 
         it('loads widget', () => {
             expect(klarnaCredit.init).toHaveBeenCalledWith({ client_token: 'foo' });
+            expect(klarnaCredit.load)
+                .toHaveBeenCalledWith({ container: '#container' }, {}, expect.any(Function));
             expect(klarnaCredit.load).toHaveBeenCalledTimes(1);
         });
 
+        it('loads widget in EU', async () => {
+            store = createCheckoutStore(getCheckoutEUStoreState());
+            strategy = new KlarnaPaymentStrategy(
+                store,
+                orderActionCreator,
+                paymentMethodActionCreator,
+                remoteCheckoutActionCreator,
+                scriptLoader
+            );
+            jest.spyOn(store, 'dispatch').mockReturnValue(Promise.resolve(store.getState()));
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(paymentMethodMock);
+
+            await strategy.initialize({ methodId: paymentMethod.id, klarna: { container: '#container', onLoad } });
+
+            expect(klarnaCredit.load)
+                .toHaveBeenCalledWith({ container: '#container' }, getKlarnaUpdateSessionParams(), expect.any(Function));
+        });
+
         it('triggers callback with response', () => {
             expect(onLoad).toHaveBeenCalledWith({ show_form: true });
         });
+
+        it('throws error if required data is not loaded', async () => {
+            try {
+                store = createCheckoutStore();
+                strategy = new KlarnaPaymentStrategy(
+                    store,
+                    orderActionCreator,
+                    paymentMethodActionCreator,
+                    remoteCheckoutActionCreator,
+                    scriptLoader
+                );
+
+                await strategy.initialize({ methodId: paymentMethod.id, klarna: { container: '#container', onLoad } });
```

Comment:
You only need the `try` for `initialize`

```ts
try {
	await strategy.initialize({ methodId: paymentMethod.id, klarna: { container: '#container', onLoad } });
} catch (error) {
    // ...
```

Reply:
not a blocker, but a better way to test for exceptions is using 

```javascript
expect(() => await strategy.initialize({ methodId: paymentMethod.id, klarna: { container: '#container', onLoad } }))
    .toThrowError(MissingDataError)
```